test/test_nn.py

```diff
@@ -13570,6 +13570,34 @@ def test_MaxPool_zero_batch_dim(self, device):
             inp = torch.ones(1, 0, 50, 44, 31, device=device)
             mod(inp)
 
+    @onlyOnCPUAndCUDA
+    def test_MaxUnpool_zero_batch_dim(self, device):
+        pool = torch.nn.MaxPool1d(2, stride=2, return_indices=True).to(device)
+        unpool = torch.nn.MaxUnpool1d(2, stride=2).to(device)
+        inp = torch.randn(0, 10, 10, requires_grad=True, device=device)
+        output, indices = pool(inp)
+        output.requires_grad_(True)
+        unpool(output, indices).sum().backward()
```

Comment:
Should there be a check that grads are zeros as well? Same throughout

Reply:
Done.